### PR TITLE
Add removed kafka sample in ballerina in action

### PR DESCRIPTION
**Pull request by dilanSachi**
## Purpose
```ballerina
import ballerina/lang.value;
import ballerinax/kafka;

configurable string groupId = "order-consumers";
configurable string orders = "orders";
configurable string paymentSuccessOrders = "payment-success-orders";
configurable decimal pollingInterval = 1;
configurable string kafkaEndpoint = kafka:DEFAULT_URL;

type Order readonly & record {|
    int id;
    string desc;
    PaymentStatus paymentStatus;
|};

enum PaymentStatus {
    SUCCESS,
    FAIL
}

final kafka:ConsumerConfiguration consumerConfigs = {
    groupId: groupId,
    topics: [orders],
    offsetReset: kafka:OFFSET_RESET_EARLIEST,
    pollingInterval: pollingInterval
};

service on new kafka:Listener(kafkaEndpoint, consumerConfigs) {
    private final kafka:Producer orderProducer;

    function init() returns error? {
        self.orderProducer = check new (kafkaEndpoint);
    }

    remote function onConsumerRecord(kafka:ConsumerRecord[] records) returns error? {
        check from var {value} in records
            let var orderString = check string:fromBytes(value)
            let Order 'order = check value:fromJsonStringWithType(orderString)
            where 'order.paymentStatus == SUCCESS
            do {
                kafka:ProducerRecord successfulOrder = {
                    topic: paymentSuccessOrders,
                    value: 'order.toString().toBytes()
                };
                check self.orderProducer->send(successfulOrder);
            };
    }
}


```
> Fixes #

## Check List

- [ ] **Page Addition**
  - [ ] Add `permalink` to pages
  - [ ] If contains empty folder(s), Add front-matter `redirect_to:`

- [ ] **Page Rename**
  - [ ] Add front-matter `redirect_from`
  - [ ] Add front-matter `redirect_to:` (If applicable)

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -353,6 +354,69 @@ <h2>Ballerina in Action</h2>
...
                </div>
             </div>
+            <div id="fifthTab" class="tab-pane fade"  >
+            <div class="row cdisplayFlex" style="background-color: #f5f5f5;">
+                <div class="col-lg-12 col-md-12 col-sm-12 focusPane" style="padding:20px;border: 1px #c7c7c7 solid;">
+                    <div class="FullCode" style="background-color: #f5f5f5;">
+                        <div class="highlight">
+                            <pre class="language-ballerina basic ballerina">
+                                <code class="language-ballerina ballerina" style="font-size: 12px !important;">
+import ballerinax/kafka;
```
Comment (sameerajayasoma) [inline index.html:364]: Let's swap the order of these imports? I would use `Optimize imports` action in our VSCode plugin

File: index.html [hunk 2]
```diff
@@ -353,6 +354,69 @@ <h2>Ballerina in Action</h2>
...
+import ballerina/lang.value;
+
+configurable string groupId = "group-id";
+configurable string listeningTopic = "topic-orders";
```
Comment (sameerajayasoma) [inline index.html:368]: I would use meaningful topic names instead of `listeningTopic` and `publishingTopic`.

File: index.html [hunk 3]
```diff
@@ -353,6 +354,69 @@ <h2>Ballerina in Action</h2>
...
+
+final kafka:ConsumerConfiguration consumerConfigs = {
+    groupId: groupId,
+    topics: [listeningTopic],
+    offsetReset: kafka:OFFSET_RESET_EARLIEST,
+    pollingInterval: pollingInterval
+};
+
+service on new kafka:Listener(kafkaEndpoint, consumerConfigs) {
+
```
Comment (sameerajayasoma) [inline index.html:392]: Let's remove this empty line here

File: index.html [hunk 4]
```diff
@@ -353,6 +354,69 @@ <h2>Ballerina in Action</h2>
...
+    private kafka:Producer kafkaProducer;
```
Comment (sameerajayasoma) [inline index.html:393]: Let's use a better variable name that matches with the topic that we are publishing to.

File: index.html [hunk 5]
```diff
@@ -353,6 +354,69 @@ <h2>Ballerina in Action</h2>
...
+
+    function init() returns error? {
+        self.kafkaProducer = check new (kafkaEndpoint);
+    }
+
+    remote function onConsumerRecord(kafka:ConsumerRecord[] records) returns error? {
+        check from var {value} in records
+            let var orderString = check string:fromBytes(value)
+            let Order 'order = check value:fromJsonStringWithType(orderString, Order)
```
Comment (sameerajayasoma) [inline index.html:402]:
```suggestion
            let Order 'order = check value:fromJsonStringWithType(orderString)
```

We don't need the parameter `Order` here. It should be inferred from the LHS.

File: index.html [hunk 6]
```diff
@@ -353,6 +354,69 @@ <h2>Ballerina in Action</h2>
...
+import ballerinax/kafka;
+import ballerina/lang.value;
+
+configurable string groupId = "group-id";
+configurable string listeningTopic = "topic-orders";
+configurable string publishingTopic = "topic-success-orders";
+configurable decimal pollingInterval = 1;
+configurable string kafkaEndpoint = kafka:DEFAULT_URL;
+
+type Order readonly & record {|
```
Comment (sameerajayasoma) [inline index.html:373]: Can we improve the field names to match the real-world purchase order fields? 

type Order readonly & record {|
    int id;
    string desc;
    PaymentStatus paymentStatus;
|};

**Review by sameerajayasoma — CHANGES_REQUESTED** [review]

File: index.html (hunk 6 above)
Comment (dilanSachi) [reply index.html:373]: Added in https://github.com/ballerina-platform/ballerina-dev-website/pull/3735/commits/76f775c43a7430c59f572264b8444345bf1687c8

File: index.html (hunk 5 above)
Comment (dilanSachi) [reply index.html]: Added in 76f775c

File: index.html (hunk 4 above)
Comment (dilanSachi) [reply index.html]: Added in 76f775c

File: index.html (hunk 3 above)
Comment (dilanSachi) [reply index.html]: Added in 76f775c

File: index.html (hunk 2 above)
Comment (dilanSachi) [reply index.html]: Added in 76f775c

File: index.html (hunk 1 above)
Comment (dilanSachi) [reply index.html]: Added in 76f775c

**Review by sameerajayasoma — APPROVED** [review]

**Pull request merged**